src/watchers/indexable-meta.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * WordPress Post Meta watcher.
+ *
+ * @package Yoast\YoastSEO\Watchers
+ */
+
+namespace Yoast\YoastSEO\Watchers;
+
+use Yoast\YoastSEO\Exceptions\No_Indexable_Found;
+use Yoast\YoastSEO\Models\Indexable_Meta as Indexable_Meta_Model;
+use Yoast\YoastSEO\Yoast_Model;
+
+/**
+ * Fills the Indexable Meta according to Post data.
+ */
+class Indexable_Meta {
+
+	/** @var int */
+	protected $indexable_id;
+
+	/**
+	 * Indexable_Post_Meta constructor.
+	 *
+	 * @param int $indexable_id The indexable id.
+	 */
+	public function __construct( $indexable_id ) {
+		$this->indexable_id = $indexable_id;
+	}
+
+	/**
+	 * Sets specific meta data for an indexable.
+	 *
+	 * @param string $meta_key   The key to set.
+	 * @param string $meta_value The value to set.
+	 *
+	 * @return void
+	 */
+	public function set_meta( $meta_key, $meta_value  ) {
+		try {
+			$indexable_meta = $this->get_indexable_meta( $meta_key );
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+
+		$indexable_meta->meta_key = $meta_key;
+		$indexable_meta->meta_value = $meta_value;
+		$indexable_meta->save();
+	}
+
+	/**
+	 * Fetches the indexable meta for a metafield and indexable.
+	 *
+	 * @param string $meta_key    The meta key to get object for.
+	 * @param bool   $auto_create Optional. Create the indexable if it does not exist.
+	 *
+	 * @return Indexable_Meta_Model
+	 *
+	 * @throws No_Indexable_Found Exception when no Indexable entry could be found.
+	 */
+	protected function get_indexable_meta( $meta_key, $auto_create = true ) {
+		$indexable_meta = Yoast_Model::of_type( 'Indexable_Meta' )
+		                        ->where( 'indexable_id', $this->indexable_id )
```

Comment:
Could also be nice to have a static method on `Indexable_Meta` to find one based on an `indexable_id` and a `meta_key`.